File: backend/src/Domain/Entities/Reservation.cs
using System;

namespace Domain.Entities
{
    public class Reservation
    {
        public Guid Id { get; set; }
        public Guid RestaurantId { get; set; }
        public string FirstName {get; set;}
        public string LastName {get; set;}
        public string PhoneNumber {get; set;}
        public string Email {get; set;}
        public DateTime DateTime { get; set; }
        public int People { get; set; }
        public string Occasion { get; set; }
        public string SpecialRequest { get; set; }
        public bool IsUser {get; set;}
        public Status Status {get; set;}
    }
}

File: frontend/src/app/models/reservation.ts
export interface IReservation {
    id: string,
    restaurantId: string,
    firstName: string,
    lastName: string,
    phoneNumber: string,
    email: string,
    dateAndTime: Date,
    people: number,
    occasion: string,
    specialRequest: string,
    status: IStatus
}

export enum IStatus {
    Pending,
    Approved,
    Cancelled
  }
